web/src/main/java/com/navercorp/pinpoint/web/service/AdminServiceImpl.java

```diff
@@ -147,30 +177,44 @@ private List<String> filterInactiveAgents(List<String> agentIds, int durationDay
             return Collections.emptyList();
         }
 
-        Range fastRange = Range.newRange(TimeUnit.HOURS, 1, System.currentTimeMillis());
+        return agentIds.stream()
+                .filter(agentId -> isInactiveAgent(agentId, durationDays))
+                .collect(Collectors.toList());
+    }
 
-        Calendar cal = Calendar.getInstance();
-        cal.add(Calendar.DATE, durationDays * -1);
-        final long fromTimestamp = cal.getTimeInMillis();
-        Range queryRange = Range.between(fromTimestamp, fastRange.getFrom() + 1);
+    private boolean isInactiveAgent(String agentId, int durationDays) {
+        long now = System.currentTimeMillis();
+        Range range = Range.between(now - TimeUnit.DAYS.toMillis(durationDays), now);
 
-        List<String> inactiveAgentIds = new ArrayList<>();
-        for (String agentId : agentIds) {
-            // FIXME This needs to be done with a more accurate information.
-            // If at any time a non-java agent is introduced, or an agent that does not collect jvm data,
-            // this will fail
+        return !this.agentInfoService.isActiveAgent(agentId, range);
+    }
 
-            boolean dataExists = agentInfoService.isActiveAgent(agentId, fastRange);
-            if (dataExists) {
-                continue;
+    @NotThreadSafe
+    private static class BulkConsumer<T> implements Consumer<T> {
```

Comment:
It seems that the code is difficult to understand because consumer has another consumer.